### DISCUSSION TITLE
Setting origin and destination in visitation

### DISCUSSION THREAD

**Discussion by slamander** [andrewmarx/samc#21] [Q&A]
Hi Andrew & SAMC community! 
I love the package, and I've been using a lot lately!
So, I've been working with samc's `visitation` in an application to map connectivity between two discrete locations, and I've run into persistent issues when attempting to set an origin location. In my application, I'm trying to model bi-directional visitation between two polygons (the first being a national park's boundary, and the second a small national wildlife refuge nearby). When attempting to set the national park as an origin, I've tried to supply cell numbers for all the pixels inside the park boundary but I always get the same error: 
```
Error: One or more coordinates do not correspond to non-NA cells.
```
I've double-checked (a million times) that each cell I've attempted to pass to the `origin` argument in `samc::visitation` is non-NA, doing so by checking the corresponding value from the raster layer, and despite all points being non-NA the error is still thrown. 
In fact, the only way I can successfully set an origin location is by taking a single random point from the center of the park boundary and then passing it to `samc::locate`. However, I can only escape the above error by setting a specific seed for the random 1-point sample, otherwise the same error is thrown. Ideally, I would like to set every pixel within the park boundary as an origin, but I believe this is only achievable when a destination is also set. In my case, I've set the anticipated destination as the only non-zero absorption cells. Should I also set this to the `dest` argument in the function? 
More generally, do you have any recommendations for "best practices" when setting origins and destinations in `samc::visitation`?
Thanks, 
-Alex.

**Comment by andrewmarx** [comment]
are you using `raster::extract()` with your park polygon? If so, it won't generally give results compatible with samc objects.

Assuming your park polygon is a `SpatialPolygons` or `sf` object, you should just be able to do `samc::locate(samc_obj, park_polygon)` to get the correct locations. But I guess I haven't really tested this thoroughly, so let me know if it doesn't work. Just be aware the results will probably be in a list and will need to be converted to a vector. Additionally, if your park is made up of multiple polygons, then you might get duplicate locations where two or more polygons interact with the same cell

I'm not sure how to answer the second part; I guess my question would be: if you supply multiple origin locations (your entire park) and no destination, what would you want or expect the result to look like:
- A list with a vector for each origin point (each vector would have one value for every cell in the landscape)?
- A vector that somehow aggregates the vectors in the list so that you end up with just a single vector that has one value for every cell?
- Something else?

**Comment by slamander** [comment]
@andrewmarx, 
Thanks for the rapid response! I've been trying a few different options to pass cells to `samc::locate`, including using `extract` paired with `raster::xyFromCell` to use `locate`, but I've since moved to use polygons directly (as you suggested). I've tried what you suggested and was able to escape the error! One thing I found was that my `origin` and `dest` parameters needed to have the same number of locations, which makes sense, and so I've decided to take a random sample of a matching size from each. 
As for my second question, I guess I didn't have a solid idea of what I was anticipating object-class-wise. Basically, I don't know what the difference between a) setting a desired destination as the only non-absorption location and b) setting a desired destination as the `dest` in `samc::visitation`.
